### ISSUE TITLE
`:blank` pseudo-selector support (re form fields)

### ISSUE THREAD

**Issue by brandonmcconnell** [web-platform-tests/interop#179]
### Description

`:blank` is a pseudo-selector that allows CSS to target a form field when it is blank/unpopulated.

Quoting the spec:
>The `:blank` pseudo-class applies to user-input elements whose input value is empty (consists of the empty string or otherwise null input).

### Rationale

There's currently no viable way to target a blank form field. The most common _vanilla_ workaround for this so far is to mark the field as `required` and then check for `:valid` along with some `pattern` matching, however there are quite a few fundamental problems with that approach:
* What if there is already some non-blank-related validation logic at play which conflicts with the mere non-blank check?
* What if the form field should NOT be required but still styled differently when blank for aesthetic purposes?

More recently, another workaround for this that has been growing in popularity is `:placeholder-shown` but this doesn't appear to work when there is no `placeholder` set.

Related discussions:
* https://zellwk.com/blog/check-empty-input-css/ (great example demonstrating ALL of the above workarounds)
* https://stackoverflow.com/questions/3617020/matching-an-empty-input-box-using-css
* https://stackoverflow.com/questions/16952526/detect-if-an-input-has-text-in-it-using-css-on-a-page-i-am-visiting-and-do-no
* https://stackoverflow.com/questions/8639282/notempty-css-selector-is-not-working
* https://stackoverflow.com/questions/4717299/css-apply-style-to-empty-inputs-value
* https://stackoverflow.com/questions/45972483/css-selector-for-empty-file-input

### Specification

https://w3c.github.io/csswg-drafts/selectors-4/#blank

### Tests

Struggling to find any tests

**Comment by foolip** [comment]
I also can't find any tests. All matches for `:blank` in WPT are for `about:blank`, and typos of it fixed in https://github.com/web-platform-tests/wpt/pull/36222.

**Comment by foolip** [comment]
Implementation bugs:
https://bugs.chromium.org/p/chromium/issues/detail?id=447880
https://bugzilla.mozilla.org/show_bug.cgi?id=1197736

None found for WebKit.

**Comment by annevk** [comment]
This seems like a reasonable feature, but the HTML standard needs to define how this actually works for its various elements. As far as I can tell that work hasn't been done.

E.g., does this work for `textarea` elements? What does it mean for `<input type=range>` or `<input type=checkbox>`?

**Comment by brandonmcconnell** [comment]
@annevk Good call out! Sharing some thoughts in the related issue you opened. Thanks!

**Comment by smaug----** [comment]
Would :blank apply when placeholder is set?

**Comment by foolip** [comment]
In the State of CSS 2022 question about browser incompatibilities, there were no mentions of `:blank`.

(I'm not commenting on all proposals where there _wasn't_ any signal from State of CSS, but thought I would here because it's similar to https://github.com/web-platform-tests/interop/issues/180 where I did comment.)

**Comment by foolip** [comment]
In the [MDN short survey on CSS & HTML](https://github.com/web-platform-tests/interop/issues/245), "CSS `:blank` and `:empty` pseudo-classes" was selected by ~19% of survey takers, putting it in the middle third of the 20 options. (There is [some uncertainty](https://github.com/web-platform-tests/interop/issues/245#issuecomment-1311874383) as with any survey data.)

Since `:blank` and `:empty` were bundled we can't say which of them resonated more. (This was done to reduce the total number of options.)

**Comment by brandonmcconnell** [comment]
@foolip @nairnandu Could we look into resurrecting this one for Interop 2024 as well?

**Comment by annevk** [comment]
I'm not convinced we have seen enough progress on https://github.com/w3c/csswg-drafts/issues/1967 and https://github.com/whatwg/html/issues/8451.

**Comment by brandonmcconnell** [comment]
@annevk Fair point. https://github.com/whatwg/html/issues/8451 seems pretty thorough to be, but https://github.com/w3c/csswg-drafts/issues/1967 may still require further discussion.

**Comment by kerryj89** [comment]
This would be my Interop 2026 vote.  Are we able to reopen or do I need to create a new one (if we have enough info now)?  We got so many cool advanced features, this just feels like one that would have already been done.  But it's one I've needed every year.

**Comment by benface** [comment]
Yeah this one is getting a bit ridiculous! Too late for Interop 2026 though, but please can we try to have it ready for 2027?

**Comment by annevk** [comment]
The issues are still unresolved unfortunately.

**Comment by benface** [comment]
Yes sorry, to be clear that's what I meant by "have it ready" – can we please try to resolve those issues.
